// Protocol messages describing source files.
//
// Copyright 2019-2020 the ProGraML authors.
//
// Contact Chris Cummins <chrisc.101@gmail.com>.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package programl;

option go_package = "programlpb";
option java_multiple_files = true;
option java_outer_classname = "SrcProto";
option java_package = "com.programl";

message SourceFile {
  // The source programming language.
  enum Language {
    UNKNOWN = 0;
    C = 1;
    CXX = 2;
    OPENCL = 3;
    SWIFT = 4;
    HASKELL = 5;
    FORTRAN = 6;
  }
  Language language = 2;
  string relpath = 1;
  string text = 3;
}
